components/hdw-bzr/hdw-bzr.c

```diff
@@ -567,3 +567,42 @@ void bzrResume(void)
         }
     }
 }
+
+/**
+ * @brief Save the state of the buzzer so that it can be restored later, perhaps
+ * after playing a different sound.
+ *
+ * @return A void-pointer which can be passed back to bzrRestore()
+ */
+void* bzrSave(void)
+{
+    bzrPause();
+
+    bzrTrack_t* result = malloc(sizeof(bzrTrack_t) * NUM_BUZZERS * 2);
+    for (uint16_t bIdx = 0; bIdx < NUM_BUZZERS; bIdx++)
+    {
+        memcpy(result + bIdx * 2, &buzzers[bIdx].bgm, sizeof(bzrTrack_t));
```

Comment:
I'm not the biggest fan of pointer arithmetic. `&result[bIdx * 2]` reads clearer to me. Same goes for the line below and in `bzrRestore()`.

Reply:
Oh, this returns `void*` because `bzrTrack_t` isn't defined in the header. Nice workaround.